geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/RebalanceCommand.java

```diff
@@ -79,34 +79,38 @@ public Result rebalance(
 
     ExecutorService commandExecutors =
         LoggingExecutors.newSingleThreadExecutor("RebalanceCommand", false);
-    List<Future<Result>> commandResult = new ArrayList<>();
-    Result result;
+    List<Future<ResultModel>> commandResult = new ArrayList<>();
+    ResultModel result;
     try {
       commandResult.add(commandExecutors.submit(new ExecuteRebalanceWithTimeout(includeRegions,
           excludeRegions, simulate, (InternalCache) getCache())));
 
-      Future<Result> fs = commandResult.get(0);
+      Future<ResultModel> fs = commandResult.get(0);
       if (timeout > 0) {
         result = fs.get(timeout, TimeUnit.SECONDS);
       } else {
         result = fs.get();
       }
     } catch (TimeoutException timeoutException) {
-      result = ResultBuilder.createInfoResult(CliStrings.REBALANCE__MSG__REBALANCE_WILL_CONTINUE);
+      result = ResultModel.createInfo(CliStrings.REBALANCE__MSG__REBALANCE_WILL_CONTINUE);
+    }
+
+    // if the result contains only error section, then we need to mark the result as error status
+    if (result.getSection("error") != null && result.getSectionSize() == 1) {
```

Comment:
This seems inconsistent with our overall stance that even a single member error would result in the overall command returning an error.

Reply:
I am trying not to change the old behavior. This seems like the only way to make sure of that...

Reply:
Is this an edge case?  It would be nice if the comment gave some example of how you might end up with exactly 1 error section and no other sections, as well as mention other path(s) by which error status is set and why those existing codepaths are not able to cover this one edge case.

Reply:
comments added to explain the reason to do it this way.